server/artifacts/artifact_server.go

```diff
@@ -214,6 +214,10 @@ func (a *ArtifactServer) GetArtifactFile(w http.ResponseWriter, r *http.Request)
 
 	} else { // stream the file itself
 		log.Debugf("not a directory, artifact: %+v", artifact)
+		if artifact.Deleted {
+			a.notFoundError(w)
+			return
+		}
 		err = a.returnArtifact(w, artifact, driver)
```

Comment:
Return 404 if we garbage collected the file. If for some reason, it got deleted manually outside of Argo Workflows, we could theoretically return a 404 as well, but that means needing to get the variety of error back from ArtifactDriver for each type of driver, which is more work.

Reply:
oh, I didn't realize about the wrapped errors...okay, I'll use that to get the real status I suppose